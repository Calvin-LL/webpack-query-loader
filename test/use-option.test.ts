import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe.each([4, 5] as const)('v%d "use" option', (webpackVersion) => {
  it('should work with "file-loader" value without resourceQuery', async () => {
    const compiler = getCompiler(webpackVersion, {
      use: "file-loader",
    });
    const stats = await compile(webpackVersion, compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it('should work with "file-loader" value with resourceQuery', async () => {
    const compiler = getCompiler(webpackVersion, {
      use: "file-loader",
      resourceQuery: "test",
    });
    const stats = await compile(webpackVersion, compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it('should work with "object" value with resourceQuery', async () => {
    const compiler = getCompiler(webpackVersion, {
      use: { loader: "file-loader", options: { name: "[path][name].[ext]" } },
      resourceQuery: "test",
    });
    const stats = await compile(webpackVersion, compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });
});
