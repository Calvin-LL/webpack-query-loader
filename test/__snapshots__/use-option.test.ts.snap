// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"use" option should work with "file-loader" value with resourceQuery: result 1`] = `"58468a78998d98229df03bef5df7fc4e.jpg"`;

exports[`"use" option should work with "file-loader" value without resourceQuery: result 1`] = `"58468a78998d98229df03bef5df7fc4e.jpg"`;

exports[`"use" option should work with "object" value with resourceQuery: result 1`] = `"Macaca_nigra_self-portrait_large.jpg"`;
